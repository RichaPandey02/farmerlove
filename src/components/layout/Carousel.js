//Carousel for Home Page......

import React from "react";

import carousl from "../styles/carousl.scss";


const Carousel1 = () => {
  return (
  <div className="ef-carousel">
  <input type="radio" name="unique-name" id="unique-name-1" checked class="ef-carousel__state"/>
  <input type="radio" name="unique-name" id="unique-name-2" className="ef-carousel__state"/>
  <input type="radio" name="unique-name" id="unique-name-3" className="ef-carousel__state"/>
  <div className="ef-carousel__items">
    <img src="https://www.teahub.io/photos/full/80-806201_indian-farmer-hd-wallpaper-download.jpg" alt="" class="ef-carousel__item"/>
    <img src="https://images.pexels.com/photos/6500749/pexels-photo-6500749.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940" alt=" " class="ef-carousel__item" />
    <img src="https://wallpaperaccess.com/full/803482.jpg" alt="" class="ef-carousel__item" />
  </div>
  <div className="ef-carousel__next-group">
    <label for="unique-name-2" className="ef-carousel__next">.</label>
    <label for="unique-name-3" className="ef-carousel__next">.</label>
    <label for="unique-name-1" className="ef-carousel__next">.</label>
  </div>
  <div className="ef-carousel__prev-group">
    <label for="unique-name-3" className="ef-carousel__prev">.</label>
    <label for="unique-name-2" className="ef-carousel__prev">.</label>
    <label for="unique-name-1" className="ef-carousel__prev">.</label>
  </div>
  <div className="ef-carousel__dots">
    <label for="unique-name-1" className="ef-carousel__dot">.</label>
    <label for="unique-name-2" className="ef-carousel__dot">.</label>
    <label for="unique-name-3" className="ef-carousel__dot">.</label>
  </div>
</div>
  );
};

export default Carousel1;

//"https://wallpaperaccess.com/full/4293102.jpg"