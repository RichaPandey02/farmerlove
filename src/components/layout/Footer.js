//Footer page....
import React from "react";
import Footer from "../styles/Footer.css";
const Footerpage = () => {
  return (
    <body className="footer" id="footer">
      <div className="footer__listitem">
        <h5 className="footer__listitem--heading" data-testid="getintouch">
          Get In touch
        </h5>
        <p className="footer__listitem--heading" data-testid="farmersfriend">
          Farmers Friend 👩‍👩‍👧
        </p>
        <p className="footer__listitem--heading" data-testid="Kanpur">
          Kanpur, Uttar Pradesh
        </p>
        <p className="footer__listitem--heading" data-testid="phone">
          8957907453
        </p>
        <p className="footer__listitem--heading" data-testid="email" id="mail">
          richapandey0297@gmail.com
        </p>
      </div>
    </body>
  );
};
export default Footerpage;
