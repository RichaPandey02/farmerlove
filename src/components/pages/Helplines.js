//helpline page ....

import React from "react";
import Help from "../styles/Help.css";
import { AgGridReact } from "ag-grid-react";
import "ag-grid-community/dist/styles/ag-grid.css";
import "ag-grid-community/dist/styles/ag-theme-alpine.css";
import Card from "./Card";
import Aggrid from "../styles/Aggrid.css";
import Footer from "../layout/Footer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

const Helplines = () => {
  const rowStyle = {
    background: "transparent",
    color: "#192a56",
  };

  const onGridReady = (params) => {
    
    fetch("http://localhost:3000/api/HelplineSchema")
      .then((resp) => resp.json())
      .then((resp) => params.api.applyTransaction({ add: resp }));
  };

  const columns = [
    {
      headerName: "TITLE",
      field: "title",
    },
    {
      headerName: "DETAILS",
      field: "details",
    },
  ];

  const defaultColDef = {
    sortable: true,
    editable: true,
    flex: 1,
    filter: true,
    floatingFilter: true,
    minWidth: 120,
    cellClass: "grid-cell-centered",

    rowStyle: { rowStyle },
  };

  return (
    <div>
      <Card />

      <section></section>
      <div className="about-para">
        <br />
        <h1 className="about-para__helpline" data-testid="helpline">
          Available Helplines ðŸ˜ŠðŸ˜Š
        </h1>
        <br></br>

        <h5 data-testid="helpline__testcase" className="about-para__help">
          Bharat Sarkar has started a Kisan Call Center for the support of the
          farmers.
        </h5>
      </div>
      <br></br>
      <div
        className="ag-theme-alpine manage-table
         container table-responsive-sm "
      >
        <AgGridReact
          columnDefs={columns}
          defaultColDef={defaultColDef}
          onGridReady={onGridReady}
          pagination={true}
          paginationPageSize={10}
          paginationAutoPageSize={true}
          height="350px"
          rowStyle={rowStyle}
        />
      </div>
      <br />
      <br />
      <Footer />
    </div>
  );
};

export default Helplines;
