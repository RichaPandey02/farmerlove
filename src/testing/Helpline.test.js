import Helplines from "../components/pages/Helplines";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Carousel from "../components/layout/Carousel"
import { BrowserRouter } from "react-router-dom";
import Footer from '../components/layout/Footer'
import {shallow,mount} from 'enzyme';

afterEach(() => {
  cleanup();
});

describe('shallow Helplines Page',()=>{
  it('render the class',()=>{
    let wrapper=shallow(<Helplines/>)
    console.log(wrapper.debug())
    expect(wrapper.exists('.about-para')).toEqual(true)
})
it('render the class name for paragarph',()=>{
  let wrapper=shallow(<Helplines/>)
  console.log(wrapper.debug())
  expect(wrapper).toMatchSnapshot();
})
it('snapshot for helpline',()=>{
  let wrapper=shallow(<Helplines/>)
  console.log(wrapper.debug())
  expect(wrapper.exists('.about-para__help')).toEqual(true)
})
it('render the class name for Aggrid',()=>{
  let wrapper=shallow(<Helplines/>)
  console.log(wrapper.debug())
  expect(wrapper.exists('.ag-theme-alpine')).toEqual(true)
})
})
it("demo testing", () => {
  expect(true).toBe(true);
});
test("to check whether carausel component rendered", () => {
  render(
    
        <Carousel />

  );
});  

test("to check whether Helplines component rendered", () => {
  render(
    <BrowserRouter>
      <Helplines />
    </BrowserRouter>
  );
});

it("the background color should be #192a56", () => {
  const color = "#192a56";
  expect(color).toEqual("#192a56");
});
it("the background should be", () => {
  const background = "transparent";
  expect(background).toEqual("transparent");
});
test("should render the paragraph ", () => {
  render(<Helplines />);
  const element = screen.getByTestId("helpline__testcase");
  expect(element).toBeInTheDocument();
  expect(element).not.toContainHTML("<h5>");
});

test("should render the Heading ", () => {
  render(<Helplines />);
  const element = screen.getByTestId("helpline");
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent("Available Helplines ðŸ˜ŠðŸ˜Š");
});
it("to be equal to 1", () => {
  const num = 1;
  expect(num).toEqual(1);
});

it("should be truthy", () => {
  //Assertions/Expectations
  expect(true).toBeTruthy();
});

test("to check whether Footer component rendered", () => {
  render(
    <BrowserRouter>
        <Footer />
        </BrowserRouter>
    
  );
});  

describe('Mount vs Shallow',()=>{
  it('snapshot 3',()=>{
    let wrapper=mount(<Helplines/>)
    console.log(wrapper.debug());
  })
})