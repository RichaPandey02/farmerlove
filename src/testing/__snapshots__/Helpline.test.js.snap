// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Helplines Page render the class name for paragarph 1`] = `
<div>
  <Slider />
  <section />
  <div
    className="about-para"
  >
    <br />
    <h1
      data-testid="helpline"
    >
      Available Helplines ðŸ˜ŠðŸ˜Š
    </h1>
    <br />
    <h5
      className="about-para__help"
      data-testid="helpline__testcase"
    >
      Bharat Sarkar has started a Kisan Call Center for the support of the farmers.
    </h5>
  </div>
  <br />
  <div
    className="ag-theme-alpine manage-table
 container table-responsive-sm "
  >
    <AgGridReact
      columnDefs={
        Array [
          Object {
            "field": "title",
            "headerName": "TITLE",
          },
          Object {
            "field": "details",
            "headerName": "DETAILS",
          },
        ]
      }
      defaultColDef={
        Object {
          "cellClass": "grid-cell-centered",
          "editable": true,
          "filter": true,
          "flex": 1,
          "floatingFilter": true,
          "minWidth": 120,
          "rowStyle": Object {
            "rowStyle": Object {
              "background": "transparent",
              "color": "#192a56",
            },
          },
          "sortable": true,
        }
      }
      disableStaticMarkup={false}
      height="350px"
      legacyComponentRendering={false}
      maxComponentCreationTimeMs={1000}
      onGridReady={[Function]}
      pagination={true}
      paginationAutoPageSize={true}
      paginationPageSize={10}
      rowStyle={
        Object {
          "background": "transparent",
          "color": "#192a56",
        }
      }
    />
  </div>
  <br />
  <br />
  <Footerpage />
</div>
`;
